### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by Barben360** [andyholmes/gnome-shell-extension-gsconnect#330]
After a moment, process `gjs` is constantly taking 100% of one CPU thread. When killing it, it relaunches gsconnect so I guess this is related.
Any idea?

**Comment by andyholmes** [comment]
Are there any errors in GNOME Logs for that time? GSConnect is a `gjs` process, but I'm not aware of it causing high CPU usage, except in the case of system proxies.

**Comment by andyholmes** [comment]
Are you experiencing high CPU usage for `daemon.js` or for `gnome-shell`?

The log you posted are assertion warnings from gnome-shell that shouldn't have any side-effects other than printing to the log. If they are, you should open a new issue for that.

To view the logs for the GSConnect service (`daemon.js`) you should inspect `journalctl /usr/bin/gjs -f`.

**Comment by Znert** [comment]
I'm sorry, the command I posted was the first thing when googling for getting gnome logs.

Currently, I don't experience the issue, so I can't give you details an what causes it exactly (will follow up once I get the spike again), but here is (part of) my log when running the command you gave me:

<details><summary>Logs from 19th to 20th of November</summary><p>
```
-- Logs begin at Sat 2018-08-18 16:57:16 CEST, end at Tue 2018-11-20 21:34:57 CET. --
Nov 19 10:19:03 sprectre-fedora gjs[2996]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 10:19:03 sprectre-fedora gjs[2996]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 10:19:03 sprectre-fedora gjs[2996]: g_variant_new_string: assertion 'string != NULL' failed
Nov 19 10:31:13 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 10:31:13 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:39:07 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:39:07 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:39:12 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:39:12 sprectre-fedora gjs[2996]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                           broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                           activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                           reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:39:13 sprectre-fedora gjs[2996]: JS WARNING: [/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/core.js 169]: reference to undefined property "GSocketOutputStream"
Nov 19 20:39:13 sprectre-fedora gjs[2996]: WARNING: znert-manjaro: Unsupported packet type (kdeconnect.mousepad.keyboardstate)
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: Failed to load shared library 'libgdprivate-1.0.so' referenced by the typelib: libgdprivate-1.0.so: cannot open shared object file: No such file or directory
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: libgdprivate-1.0.so: cannot open shared object file: No such file or directory
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 20:39:20 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 20:46:29 sprectre-fedora gjs[9060]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:46:29 sprectre-fedora gjs[9060]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:46:29 sprectre-fedora gjs[9060]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           _onDisabledPlugins/<@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:760:29
                                           _onDisabledPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:760:9
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:46:29 sprectre-fedora gjs[9060]: JS ERROR: WaylandNotSupported
                                           _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                           loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                           _onDisabledPlugins/<@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:760:29
                                           _onDisabledPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:760:9
                                           @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 20:46:29 sprectre-fedora gjs[9060]: g_variant_new_string: assertion 'string != NULL' failed
Nov 19 20:46:30 sprectre-fedora gjs[9060]: [113B blob data]
Nov 19 20:46:30 sprectre-fedora gjs[9060]: [53B blob data]
Nov 19 20:46:30 sprectre-fedora gjs[9060]: JS WARNING: [/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/contacts.js 163]: reference to undefined property "ENCODING"
Nov 19 20:55:53 sprectre-fedora gjs[9060]: [113B blob data]
Nov 19 20:55:53 sprectre-fedora gjs[9060]: [53B blob data]
Nov 19 21:00:29 sprectre-fedora gjs[11291]: JS ERROR: WaylandNotSupported
                                            _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                            loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                            loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 21:00:29 sprectre-fedora gjs[11291]: JS ERROR: WaylandNotSupported
                                            _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                            loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                            loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 21:00:29 sprectre-fedora gjs[11291]: g_variant_new_string: assertion 'string != NULL' failed
Nov 19 21:00:36 sprectre-fedora gjs[11291]: WARNING: Simons Nokia: sftp: fuse: mountpoint is not empty
Nov 19 21:00:36 sprectre-fedora gjs[11291]: WARNING: Simons Nokia: sftp: fuse: if you are sure this is safe, use the 'nonempty' mount option
Nov 19 21:12:45 sprectre-fedora gjs[11291]: JS WARNING: [/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/core.js 169]: reference to undefined property "GSocketOutputStream"
Nov 19 21:15:12 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:15:12 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:15:12 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:15:12 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:18:08 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:18:08 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:18:08 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:18:08 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:18:28 sprectre-fedora gjs[15239]: JS ERROR: WaylandNotSupported
                                            _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                            loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                            loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 21:18:28 sprectre-fedora gjs[15239]: JS ERROR: WaylandNotSupported
                                            _init@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:112:21
                                            loadPlugin@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:772:26
                                            loadPlugins@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:797:19
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 19 21:18:28 sprectre-fedora gjs[15239]: g_variant_new_string: assertion 'string != NULL' failed
Nov 19 21:18:29 sprectre-fedora gjs[15239]: WARNING: znert-manjaro: Unsupported packet type (kdeconnect.mousepad.keyboardstate)
Nov 19 21:18:29 sprectre-fedora gjs[15239]: [113B blob data]
Nov 19 21:18:29 sprectre-fedora gjs[15239]: [53B blob data]
Nov 19 21:19:55 sprectre-fedora gjs[15239]: GtkDialog mapped without a transient parent. This is discouraged.
Nov 19 21:19:59 sprectre-fedora gjs[15239]: WARNING: znert-manjaro: ID_REGEX.exec(...) is null
Nov 19 21:20:03 sprectre-fedora gjs[15239]: WARNING: znert-manjaro: ID_REGEX.exec(...) is null
Nov 19 21:21:49 sprectre-fedora gjs[15239]: GtkDialog mapped without a transient parent. This is discouraged.
Nov 19 21:25:53 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:25:53 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:25:53 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:25:53 sprectre-fedora gnome-documents[6406]: JS ERROR: GLib.Error g-invoke-error-quark: Could not locate gd_filename_strip_extension: (null)
Nov 19 21:32:45 sprectre-fedora gjs[15239]: [113B blob data]
Nov 19 21:32:45 sprectre-fedora gjs[15239]: [53B blob data]
Nov 19 21:35:29 sprectre-fedora gjs[15239]: JS WARNING: [/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/core.js 169]: reference to undefined property "GSocketOutputStream"
Nov 19 21:35:29 sprectre-fedora gjs[15239]: WARNING: znert-manjaro: Unsupported packet type (kdeconnect.mousepad.keyboardstate)
Nov 19 22:44:05 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:44:05 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:45:26 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:45:26 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:45:32 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:45:32 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:45:54 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:50:25 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:50:49 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:51:13 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:51:15 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:51:37 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:51:39 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:57:22 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:57:24 sprectre-fedora gjs[15239]: gvc_mixer_stream_get_port: assertion 'stream->priv->ports != NULL' failed
Nov 19 22:58:15 sprectre-fedora gjs[15239]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                            broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                            _onNetworkChanged@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:69:13
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 20 21:18:05 sprectre-fedora gjs[15239]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                            broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                            broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                            activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                            reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
Nov 20 21:18:10 sprectre-fedora gjs[15239]: JS ERROR: Gio.IOErrorEnum: Error sending message: Network is unreachable
                                            broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/lan.js:280:13
                                            broadcast@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:185:21
                                            activate@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:348:13
                                            reconnect@/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:207:17
                                            @/home/znert/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:881:2
```
End of log
</p></details>

**Comment by andyholmes** [comment]
Some of those errors should now be silenced in `master`, you might want to install this Zip:

[gsconnect@andyholmes.github.io.zip](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/2601855/gsconnect.andyholmes.github.io.zip)

There's actually a _whole_ lot of fixes in there that haven't made it to extensions.gnome.org. There are instructions in the Wiki for [Installing from Zip](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Installation#install-from-zip) if that's something you're up for, I'd recommend it.

**Comment by Dolland** [comment]
Also i have this kind of problem when listening to music with Audacious.
I'm on Ubuntu 18.10
There is a screenshot of htop when listening to music:
![image](https://user-images.githubusercontent.com/45694266/49658885-5627c580-fa43-11e8-850b-bf3dc37b435d.png)

And there is what "journalctl /usr/bin/gjs -f" returns:
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
dic 07 17:13:23 marco-Swift-SF314-51 gjs[1499]: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

**Comment by andyholmes** [comment]
> Also i have this kind of problem when listening to music with Audacious.

I'm not so familiar with audacious, but it seems like the MPRIS plugin might be emitting position changes very frequently (https://github.com/audacious-media-player/audacious-plugins/blob/master/src/mpris2/plugin.cc#L170-L183):

```cpp
static void update (void * object)
{
    int64_t pos = 0;
    int vol = 0;

    if (aud_drct_get_playing () && aud_drct_get_ready ())
        pos = (int64_t) aud_drct_get_time () * 1000;

    vol = aud_drct_get_volume_main ();

    g_signal_handlers_block_by_func (object, (void *) volume_changed, nullptr);
    g_object_set ((GObject *) object, "position", pos, "volume", (double) vol / 100, nullptr);
    g_signal_handlers_unblock_by_func (object, (void *) volume_changed, nullptr);
}
```

And (https://github.com/audacious-media-player/audacious-plugins/blob/master/src/mpris2/plugin.cc#L360):

```cpp
timer_add (TimerRate::Hz4, update, object_player);
```

Compare this with the [MPRIS Specification](https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Property:Position) which says:

> The org.freedesktop.DBus.Properties.PropertiesChanged signal is not emitted when this property changes. 

Probably you should file an upstream report with audacious for this, and ask if these signals are being emitted over DBus.

**Comment by ferdnyc** [comment]
Audacity does indeed output `PropertiesChanged` updates on `Position` at the rate of approx. four per second when playing, on my system. Compared to `mpdris2` (the MPRIS2 monitor for the [MPD Music Player Daemon](https://github.com/MusicPlayerDaemon/MPD), and VLC, which only outputs messages for events like start, stop, seek, and song-change.

**Comment by andyholmes** [comment]
Ah, yeah that's way to often, especially since it's actually microsecond measurement. The high cpu with GSConnect is because it's relaying each change to connected devices.

Generally only the `Seeked()` signal should be emitted and MPRIS clients should run their own "local" timer starting from `Position`, otherwise this kind of performance issue will happen for all clients. I'd label this upstream, but I'm not sure this is related to the original report.

**Comment by ferdnyc** [comment]
> I'd label this upstream, but I'm not sure this is related to the original report.

Agreed. I'm going to take advantage of GitHub's (relatively) new "Open new issue" option for issue comments, to peel @Dolland 's comment off as a separate issue of its own, and label _that_ one "upstream". I think it's prudent to file a bug with Audacious about this, referencing our bug. (I'll also start on that.)

**Comment by andyholmes** [comment]
Can you provide any logs from `journalctl` or GNOME Logs from that period that might indicate what the problem is?

**Comment by bjurr** [comment]
Ok, think I found something. Someone else at work has GSConnect installed, and I would see his laptop as one of the advertised device. Whenever I had the lockup, I would see some message related to this device trying to connect.

I tried removing the device from the list, but it would reappear on restart. I finally turned off "Discoverable", and removed the device manually using dconf-editor. No problems since.

I'm a bit confused though as why I would see someone else's laptop running GSConnect as being a device I should be able to connect to .... doesn't really make sense, no?

**Comment by andyholmes** [comment]
Yep, you can connect any device running a KDE Connect implementation to any other. Android, Sailfish, KDE Connect, GSConnect; they can all connect to each other.

**Comment by dgel** [comment]
Hey, I'm experiencing this issue as well. I thought updating to the latest version would fix things, but it hasn't. Attaching the log from when it happened to me just now

[gsconnecterrors.txt](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/2741875/gsconnecterrors.txt)

**Comment by ferdnyc** [comment]
@dgel 
> ene 09 17:59:43 douwlaptop gjs[12970]: WARNING: ValueError: Namespace Folks not available

That error makes me think that `libfolks` isn't present on your system (GSConnect uses it to pull in contacts from the local address book) — can you check whether [the appropriate Folks packages for your distro](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Installation#desktop-contacts) are installed, and (if not) whether installing them makes a difference in the CPU usage?

**Comment by andyholmes** [comment]
@dgel It seems like from the log file you posted that you have installed this previously from GNOME Software:

    gnome-software[4301]: State change on user/*/*/shell-extension/gsconnect_andyholmes.github.io/* from available to installed is not OK
    ...
    ureadahead:/home/myuser/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/ui/packagekit.js: No such file or directory

This seems to indicate GNOME Software is interfering with the install process somehow. I would recommend you completely remove GSConnect using whatever means you originally installed it, then reinstall from this Zip:

[gsconnect@andyholmes.github.io.zip](https://github.com/andyholmes/gnome-shell-extension-gsconnect/files/2742700/gsconnect.andyholmes.github.io.zip)

    rm -rf ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io
    unzip -o gsconnect.andyholmes.github.io.zip -d ~/.local/share/gnome-shell/extensions/gsconnect.andyholmes.github.io

You should then restart GNOME Shell.

**Comment by rockorequin** [comment]
Fwiw, I just started seeing this issue on Ubuntu 19.04. Re-installing from the aforementioned zipfile fixes the issue, but gnome-shell-extensions then prompts me to upgrade the extension (which then gives high CPU usage again).

**Comment by Drawey** [comment]
Works for me too

**Comment by rockorequin** [comment]
I'm seeing this 100% CPU issue with gjs again with the latest gsconnect from git and gjs 1.55.92. It happened just now and in the gjs log there are many, many messages with the same timestamp saying:

```
JS ERROR: Error: 0x40000020 is not a valid value for flags GSocketMsgFlags
_onIncomingIdentity@/home/rocko/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/protocol/lan.js:199:24
@/home/rocko/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:744:2
```

**Comment by andyholmes** [comment]
> JS ERROR: Error: 0x40000020 is not a valid value for flags GSocketMsgFlags

This is related to #468 and should be fixed in [glib !710](https://gitlab.gnome.org/GNOME/glib/merge_requests/710) and [gobject-instrospection !141](https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/141).

**Comment by rockorequin** [comment]
If I have understood correctly, both https://gitlab.gnome.org/GNOME/glib/merge_requests/710 and https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/141 are mainly about removing an extra semicolon from the comment for g_socket_receive_messages_with_timeout, ie from "* @flags: (type GSocketMsgFlags): (inout):", aren't they? (Do the comments get parsed to figure out the arguments that get passed, then?) Or are there more changes that I have missed?

I checked gobject-introspection 1.60.0-1 and that semicolon is already removed; it's still present in libglib2 2.60.0-1, so I built and installed a version without the semi-colon, but I'm still seeing the 100% CPU issue in gjs, so I don't understand what's going on.

Quite apart from the bug in glib, though, isn't the gjs 100% CPU usage issue problem caused because gsconnect calls g_socket_receive_messages_with_timeout, which fails immediately, so gsconnect continually calls it? Is there an error returned that gsconnect could perhaps check for and report its own error rather than causing the 100% CPU problem?

**Comment by ferdnyc** [comment]
> If I have understood correctly, both https://gitlab.gnome.org/GNOME/glib/merge_requests/710 and https://gitlab.gnome.org/GNOME/gobject-introspection/merge_requests/141 are mainly about removing an extra semicolon from the comment for g_socket_receive_messages_with_timeout, ie from "* @flags: (type GSocketMsgFlags): (inout):", aren't they? (Do the comments get parsed to figure out the arguments that get passed, then?) Or are there more changes that I have missed?

Those comments aren't comments. If you look at that file [gio-2.0.c](https://gitlab.gnome.org/GNOME/gobject-introspection/blob/912993caae1afd0cabb3218c7a7ee7a92c370ab2/gir/gio-2.0.c) in the gobject-introspection repository, the _entire_ contents of the file are nothing but those comment blocks, which contain the generated Gio introspection.

From the [GObject Introspection "Overview" doc](https://gi.readthedocs.io/en/latest/):
> GObject introspection is a middleware layer between C libraries (using GObject) and language bindings. The C library can be scanned at compile time and generate metadata files, in addition to the actual native C library. Then language bindings can read this metadata and automatically provide bindings to call into the C library.

The comment blocks **are** the metadata which defines the language bindings for Gio.

**Comment by andyholmes** [comment]
> isn't the gjs 100% CPU usage issue problem caused because gsconnect calls g_socket_receive_messages_with_timeout

No-ish. Actually all GSocket message related methods eventually call `g_socket_receive_message_with_timeout()` indirectly; they're all essentially wrappers around that core function. In GSConnect we would probably use `g_socket_receive_message_from()` to get the message and remote address in one go if we could, but the GSocket code is still pretty low-level and often requires pointers and structs that aren't really a thing in JavaScript.

We actually call `g_socket_read_message()` with the `G_SOCKET_MSG_PEEK` flag when the `G_IO_IN` condition (input available) triggers. So when we "peek" the data we're not waiting on data that can timeout, we're looking at data that is already waiting without consuming it (and ignoring it actually) so that we can grab the remote IP address as a side effect. We actually read the data later by wrapping the socket in a high-level stream.

The [code in question](https://github.com/andyholmes/gnome-shell-extension-gsconnect/blob/40ac140b08bbdacc0a42d4d5be5bd4d68d0bdef3/src/service/protocol/lan.js#L195-L217) is wrapped in a try-catch, so when the call to `g_socket_receive_message()`  throws an error, there is still data waiting so the `G_IO_IN` condition triggers again. The only alternative is GSConnect bailing completely and ceasing to work at all.

@ferdnyc is correct about how GIR data is generated. I'm unsure how flags are mapped in memory (`0x40000020`) however the GJS docs list `Gio.SocketMsgFlags.PEEK = 2`, so you could try replacing that flag constant with the integer `2`. These are actually the platform-level constants (eg. Linux, BSD), so that may work, but for that reason I'm not keen on using a literal number since this might not be `2` on non-Linux platforms like BSD.

In any case, this is a problem in the language bindings that I can't fix in GSConnect, but if you're sure you're running glib2 and gobject-introspection with those changes you should report that as soon as possible. Probably adding a comment on the closed GitLab issue from #468 or one the merge requests above would be the best way to get attention for this.

**Comment by andyholmes** [comment]
The possibility of an endless loop on G_IO_IN has been fixed in de32515. I've closed this issue since I believe it's now referred to several closed issues.

The introspection bug in glib2/gobject-introspection will still have to be fixed upstream before broadcasts can be properly received, which will probably prevent connecting to new devices.

**Comment by rockorequin** [comment]
Thanks, the latest gsconnect from git fixes the 100% cpu usage problem for me.

**Comment by lacostenycoder** [comment]
I got 100% CPU usage just now here
```
# journalctl /usr/bin/gnome-shell -n 1000 > gnome-shell.log
# I've removed many many duplicate lines where these IDs change but many lines 10-50 a shot
# example: Jun 18 11:09:56 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2680168 was not found when attempting to remove it

-- Logs begin at Tue 2021-05-11 15:10:35 EDT, end at Fri 2021-06-18 12:22:02 EDT. --
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:40 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: this.actor is null
                                                     vfunc_set_actor/this._monitorsChangedId<@resource:///org/gnome/shell/ui/layout.js:103:25
                                                     _monitorsChanged@resource:///org/gnome/shell/ui/layout.js:559:14
Jun 18 09:00:45 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2511002 was not found when attempting to remove it
Jun 18 09:00:46 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2511965 was not found when attempting to remove it
Jun 18 09:00:50 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2512865 was not found when attempting to remove it
Jun 18 09:01:00 lance-XPS-15-7590 gnome-shell[2172]: Usage of object.actor is deprecated for PanelMenuButton
                                                     get@resource:///org/gnome/shell/ui/environment.js:287:29
                                                     enable@/home/lance/.local/share/gnome-shell/extensions/darkthemetoggle@jacob.mischka.me/extension.js:37:2
                                                     _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
                                                     _enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:614:22
                                                     _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:613:37
                                                     _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:641:18
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
                                                     popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
                                                     _continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:502:30
                                                     deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:493:44
                                                     finish@resource:///org/gnome/shell/gdm/authPrompt.js:554:13
                                                     finish@resource:///org/gnome/shell/ui/unlockDialog.js:871:26
                                                     deactivate@resource:///org/gnome/shell/ui/screenShield.js:493:26
                                                     ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:110:57
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:169:19
Jun 18 09:01:00 lance-XPS-15-7590 gnome-shell[2172]: Usage of object.actor is deprecated for PanelMenuButton
                                                     get@resource:///org/gnome/shell/ui/environment.js:287:29
                                                     enable@/home/lance/.local/share/gnome-shell/extensions/darkthemetoggle@jacob.mischka.me/extension.js:38:2
                                                     _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
                                                     _enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:614:22
                                                     _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:613:37
                                                     _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:641:18
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
                                                     popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
                                                     _continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:502:30
                                                     deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:493:44
                                                     finish@resource:///org/gnome/shell/gdm/authPrompt.js:554:13
                                                     finish@resource:///org/gnome/shell/ui/unlockDialog.js:871:26
                                                     deactivate@resource:///org/gnome/shell/ui/screenShield.js:493:26
                                                     ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:110:57
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:169:19
Jun 18 09:01:01 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: Exception in callback for signal: updated: Error: Tried to construct an object without a GType; are you using GObject.registerClass() when inheriting from a GObject type?
                                                     BlurredLightbox@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/dialogs.js:166:9
                                                     ModalDialog@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/dialogs.js:302:26
                                                     PomodoroEndDialog@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/dialogs.js:583:9
                                                     _enableScreenNotification@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/extension.js:439:27
                                                     setMode@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/extension.js:127:26
                                                     enable/sessionModeUpdatedId<@/usr/share/gnome-shell/extensions/pomodoro@arun.codito.in/extension.js:582:27
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
                                                     popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
                                                     _continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:502:30
                                                     deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:493:44
                                                     finish@resource:///org/gnome/shell/gdm/authPrompt.js:554:13
                                                     finish@resource:///org/gnome/shell/ui/unlockDialog.js:871:26
                                                     deactivate@resource:///org/gnome/shell/ui/screenShield.js:493:26
                                                     ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:110:57
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:169:19
Jun 18 09:01:02 lance-XPS-15-7590 gnome-shell[2172]: Attempting to run a JS callback during garbage collection. This is most likely caused by destroying a Clutter actor or GTK widget with ::destroy signal connected, or using the destroy(), dispose(), or remove() vfuncs. Because it would crash the application, it has been blocked.
Jun 18 09:01:02 lance-XPS-15-7590 gnome-shell[2172]: The offending callback was set_actor(), a vfunc.
Jun 18 09:01:02 lance-XPS-15-7590 gnome-shell[2172]: == Stack trace for context 0x55b6072eb6a0 ==
Jun 18 09:01:08 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2520524 was not found when attempting to remove it
Jun 18 09:01:16 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2520787 was not found when attempting to remove it
Jun 18 09:01:18 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2000014 specified for 0x2000067.
Jun 18 09:01:21 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2521176 was not found when attempting to remove it
Jun 18 09:01:22 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 09:22:28 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2532985 was not found when attempting to remove it
Jun 18 09:23:12 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2539449 was not found when attempting to remove it
Jun 18 09:23:18 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 09:23:19 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 09:23:28 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2540302 was not found when attempting to remove it
Jun 18 09:23:32 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2540469 was not found when attempting to remove it
Jun 18 09:34:28 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2549558 was not found when attempting to remove it
Jun 18 09:34:40 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2e00008 specified for 0x2e00006.
Jun 18 10:03:36 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2596170 was not found when attempting to remove it
Jun 18 10:03:57 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2596181 was not found when attempting to remove it
Jun 18 10:03:58 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2e00008 specified for 0x2e00006.
Jun 18 10:04:14 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2596192 was not found when attempting to remove it
Jun 18 11:08:58 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2676560 was not found when attempting to remove it
Jun 18 11:09:04 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2676753 was not found when attempting to remove it
Jun 18 11:09:24 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 11:09:46 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2679983 was not found when attempting to remove it
Jun 18 11:09:56 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2680168 was not found when attempting to remove it
Jun 18 11:10:19 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2680933 was not found when attempting to remove it
Jun 18 11:10:27 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2681233 was not found when attempting to remove it
Jun 18 11:26:44 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x2201303
Jun 18 11:28:36 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2707632 was not found when attempting to remove it
Jun 18 11:28:40 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2707654 was not found when attempting to remove it
Jun 18 11:28:48 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2707660 was not found when attempting to remove it
Jun 18 11:28:57 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2707784 was not found when attempting to remove it
Jun 18 11:38:35 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2717107 was not found when attempting to remove it
Jun 18 11:38:37 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 11:38:40 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2e00008 specified for 0x2e00006.
Jun 18 11:38:53 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 11:38:57 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2718068 was not found when attempting to remove it
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 4
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:10 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: TypeError: malformed UTF-8 character sequence at offset 0
Jun 18 11:39:15 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 11:39:21 lance-XPS-15-7590 gnome-shell[2172]: Failed to read extents of focused component: timeout from dbind
Jun 18 11:39:28 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2722161 was not found when attempting to remove it
Jun 18 11:40:44 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2725034 was not found when attempting to remove it
Jun 18 11:41:32 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2726119 was not found when attempting to remove it
Jun 18 11:41:38 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2726209 was not found when attempting to remove it
Jun 18 11:41:44 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x220000e
Jun 18 11:41:59 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2727195 was not found when attempting to remove it
Jun 18 11:42:02 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2727298 was not found when attempting to remove it
Jun 18 11:46:01 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2739947 was not found when attempting to remove it
Jun 18 11:46:07 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2740112 was not found when attempting to remove it
Jun 18 11:53:33 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 11:53:39 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 11:53:47 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 11:53:53 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 11:54:05 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 12:10:30 lance-XPS-15-7590 gnome-shell[2172]: Usage of object.actor is deprecated for PanelMenuButton
                                                     get@resource:///org/gnome/shell/ui/environment.js:287:29
                                                     enable@/home/lance/.local/share/gnome-shell/extensions/darkthemetoggle@jacob.mischka.me/extension.js:37:2
                                                     _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
                                                     _enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:614:22
                                                     _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:613:37
                                                     _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:641:18
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
                                                     popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
                                                     _continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:502:30
                                                     deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:493:44
                                                     finish@resource:///org/gnome/shell/gdm/authPrompt.js:554:13
                                                     finish@resource:///org/gnome/shell/ui/unlockDialog.js:871:26
                                                     deactivate@resource:///org/gnome/shell/ui/screenShield.js:493:26
                                                     ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:110:57
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:169:19
Jun 18 12:10:30 lance-XPS-15-7590 gnome-shell[2172]: Usage of object.actor is deprecated for PanelMenuButton
                                                     get@resource:///org/gnome/shell/ui/environment.js:287:29
                                                     enable@/home/lance/.local/share/gnome-shell/extensions/darkthemetoggle@jacob.mischka.me/extension.js:38:2
                                                     _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:167:32
                                                     _enableAllExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:614:22
                                                     _enableAllExtensions@resource:///org/gnome/shell/ui/extensionSystem.js:613:37
                                                     _sessionUpdated@resource:///org/gnome/shell/ui/extensionSystem.js:641:18
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _sync@resource:///org/gnome/shell/ui/sessionMode.js:198:14
                                                     popMode@resource:///org/gnome/shell/ui/sessionMode.js:166:14
                                                     _continueDeactivate@resource:///org/gnome/shell/ui/screenShield.js:502:30
                                                     deactivate/<@resource:///org/gnome/shell/ui/screenShield.js:493:44
                                                     finish@resource:///org/gnome/shell/gdm/authPrompt.js:554:13
                                                     finish@resource:///org/gnome/shell/ui/unlockDialog.js:871:26
                                                     deactivate@resource:///org/gnome/shell/ui/screenShield.js:493:26
                                                     ScreenShield/</<@resource:///org/gnome/shell/ui/screenShield.js:110:57
                                                     _emit@resource:///org/gnome/gjs/modules/core/_signals.js:133:47
                                                     _convertToNativeSignal@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:169:19
Jun 18 12:10:39 lance-XPS-15-7590 gnome-shell[2172]: == Stack trace for context 0x55b6072eb6a0 ==
Jun 18 12:10:51 lance-XPS-15-7590 gnome-shell[2172]: Window manager warning: Invalid WM_TRANSIENT_FOR window 0x2e00008 specified for 0x2e00006.
Jun 18 12:11:21 lance-XPS-15-7590 gnome-shell[2172]: JS ERROR: Gio.IOErrorEnum: Timeout was reached
                                                     _addDevices/objects</<@/home/lance/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/utils/remote.js:362:45
Jun 18 12:12:23 lance-XPS-15-7590 gnome-shell[2172]: An active wireless connection, in infrastructure mode, involves no access point?
Jun 18 12:13:42 lance-XPS-15-7590 gnome-shell[2172]: Ignoring search provider /usr/share/gnome-shell/search-providers/org.gnome.Nautilus.search-provider.ini: missing DesktopId
Jun 18 12:13:52 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2752904 was not found when attempting to remove it
Jun 18 12:13:55 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2753000 was not found when attempting to remove it
Jun 18 12:14:10 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2753081 was not found when attempting to remove it
Jun 18 12:14:12 lance-XPS-15-7590 gnome-shell[2172]: ../clutter/clutter/clutter-actor.c:10558: The clutter_actor_set_allocation() function can only be called from within the implementation of the ClutterActor::allocate() virtual function.
Jun 18 12:14:43 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2754451 was not found when attempting to remove it
Jun 18 12:14:48 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2754565 was not found when attempting to remove it
Jun 18 12:14:50 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2754577 was not found when attempting to remove it
Jun 18 12:21:49 lance-XPS-15-7590 gnome-shell[2172]: Source ID 2765530 was not found when attempting to remove it


```

**Comment by stragu** [comment]
I get high CPU usage as well, I notice it because my fans keep running at high speed because of this process. (although it idles around 10%)
Is there a new report somewhere?

**Comment by andyholmes** [comment]
There is not, no.

Most of these reports seemed to be related to the bug fixed by #1233. If you are running a version that includes that fix, then you should open a new report, preferably with helpful logs or a way to reproduce the problem.

Unfortunately bug reports along the lines of "High CPU usage" tend to get hijacked or conflated with different causes, but if that's all you can figure out, that's OK.

**Comment by stragu** [comment]
Thank you, @andyholmes, I appreciate your detailed reply! I uninstalled for now but if I need GSConnect again and still see the issue, I will find some time to report with more info.
